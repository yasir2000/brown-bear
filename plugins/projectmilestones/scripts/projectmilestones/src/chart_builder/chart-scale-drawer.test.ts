/*
 * Copyright (c) Enalean, 2019 - present. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

import { addScaleLines } from "./chart-scale-drawer";
import type { Selection } from "d3-selection";
import { select } from "d3-selection";

describe("ScaleDrawer -", () => {
    describe("addScaleLines -", () => {
        it("When the scales are created, Then there are 2 line element with their attributes", () => {
            const chart_div = getDocument();

            const coordinate = {
                x_coordinate_minimum: 0,
                y_coordinate_minimum: 100,
                x_coordinate_maximum: 200,
                y_coordinate_maximum: 500,
            };

            addScaleLines(chart_div, coordinate);

            expect(chart_div).toMatchSnapshot();
        });
    });

    function getDocument(): Selection<SVGSVGElement, unknown, null, undefined> {
        const local_document = document.implementation.createHTMLDocument();
        const chart_div = local_document.createElement("svg");
        return select(chart_div).append("svg").attr("width", 100).attr("height", 100);
    }
});
