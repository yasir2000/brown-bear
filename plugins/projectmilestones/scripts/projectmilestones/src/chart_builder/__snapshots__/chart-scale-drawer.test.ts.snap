// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScaleDrawer - addScaleLines - When the scales are created, Then there are 2 line element with their attributes 1`] = `
Selection {
  "_groups": Array [
    Array [
      <svg
        height="100"
        width="100"
      >
        <line
          class="release-line-scale"
          x1="0"
          x2="210"
          y1="100"
          y2="100"
        />
        <line
          class="release-line-scale"
          x1="0"
          x2="0"
          y1="100"
          y2="490"
        />
      </svg>,
    ],
  ],
  "_parents": Array [
    null,
  ],
}
`;
